modules/carodaBidAdapter.js

```diff
@@ -0,0 +1,214 @@
+// jshint esversion: 6, es3: false, node: true
+'use strict'
+
+import { registerBidder } from '../src/adapters/bidderFactory.js'
+import { BANNER, VIDEO } from '../src/mediaTypes.js'
+import {
+  deepAccess,
+  deepSetValue,
+  logError,
+  mergeDeep,
+  parseSizesInput
+} from '../src/utils.js'
+import { config } from '../src/config.js'
+
+const { getConfig } = config
+
+const BIDDER_CODE = 'caroda'
+const GVLID = 954
+
+// some state info is required to synchronize with Caroda ad server
+const topUsableWindow = getTopUsableWindow()
+
+export const spec = {
+  code: BIDDER_CODE,
+  gvlid: GVLID,
+  supportedMediaTypes: [BANNER, VIDEO],
+  isBidRequestValid: bid => {
+    const params = bid.params || {}
+    const { ctok, placementId } = params
+    return typeof ctok === 'string' && (
+      typeof placementId === 'string' ||
+      typeof placementId === 'undefined')
+  },
+  buildRequests: (validBidRequests, bidderRequest) => {
+    const pageViewId = topUsableWindow.carodaPageViewId || Math.floor(Math.random() * 1e9)
```

Comment:
how do you expect this value to remain stable if missing?

Reply:
fixed